modules/programs/wlogout.nix

```diff
@@ -0,0 +1,160 @@
+          description = "Command to execute when clicked.";
+        };
+
+        text = mkOption {
+          type = str;
+          default = "";
+          example = "Shutdown";
+          description = "Text displayed on button.";
+        };
+
+        keybind = mkOption {
+          type = str;
+          default = "";
+          example = "s";
+          description = "Keyboard character to trigger this action.";
+        };
+
+        height = mkOption {
+          type = nullOr (numbers.between 0 1);
+          default = null;
+          example = 0.5;
+          description = "Relative height of tile.";
+        };
+
+        width = mkOption {
+          type = nullOr (numbers.between 0 1);
+          default = null;
+          example = 0.5;
+          description = "Relative width of tile.";
+        };
+
+        circular = mkOption {
+          type = nullOr bool;
+          default = null;
+          example = true;
+          description = "Make button circular.";
+        };
+      };
+    };
+in {
+  meta.maintainers = [ lib.maintainers.Scrumplex ];
+  options.programs.wlogout = with lib.types; {
+    enable = mkEnableOption "wlogout";
+
+    package = mkOption {
+      type = package;
+      default = pkgs.wlogout;
+      defaultText = literalExpression "pkgs.wlogout";
+      description = ''
+        wlogout package to use. Set to <code>null</code> to use the default package.
+      '';
+    };
+
+    layout = mkOption {
+      type = listOf wlogoutLayoutConfig;
+      default = [ ];
+      description = ''
+        Layout configuration for wlogout, see <link
+          xlink:href="https://github.com/ArtsyMacaw/wlogout#config"/>
+        for supported values.
+      '';
+      example = literalExpression ''
+        [
+          {
+            label = "shutdown";
+            action = "systemctl poweroff";
+            text = "Shutdown";
+            keybind = "s";
+          }
+        ]
+      '';
+    };
+
+    style = mkOption {
+      type = nullOr (either path str);
+      default = null;
+      description = ''
+        CSS style of the bar.
+        </para>
+        <para>
+        See <link xlink:href="https://github.com/ArtsyMacaw/wlogout#style"/>
+        for the documentation.
+        </para>
+        <para>
+        If the value is set to a path literal, then the path will be used as the css file.
+      '';
+      example = ''
+        window {
+          background: #16191C;
+        }
+
+        button {
+          color: #AAB2BF;
+        }
+      '';
+    };
+  };
+
+  config = let
+    # Removes nulls because wlogout ignores them.
+    # This is not recursive.
+    removeTopLevelNulls = filterAttrs (_: v: v != null);
+    cleanJSON = foo: toJSON (removeTopLevelNulls foo);
+
+    # wlogout doesn't want a JSON array, it just wants a list of JSON objects
+    layoutJsons = map cleanJSON cfg.layout;
+    layoutContent = concatStringsSep "\n" layoutJsons;
+
+  in mkIf cfg.enable (mkMerge [{
```

Comment:
We also need a platform restriction check, e.g.
https://github.com/nix-community/home-manager/blob/5f433eb164832fc507c3e1ba2a798f8c00578316/modules/services/keynav.nix#L12-L16